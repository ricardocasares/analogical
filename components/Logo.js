// external
import React from "react";
import { Activity } from "react-bytesize-icons";

// internal
import Link from "./Link";
import { themed } from "../providers/Theme";

export default themed(({ theme }) => (
  <div>
    <Link route="index">
      <Activity
        width={48}
        height={48}
        color={theme.logo}
        strokeWidth={"7%"}
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </Link>
  </div>
));
