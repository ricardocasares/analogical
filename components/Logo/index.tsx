import { FunctionComponent } from "react";
import styled from "@app/lib/styled";
import Box from "@app/components/Box";

export interface Logo {
  color?: string;
}

const Text = styled.span<Logo>`
  font-weight: 500;
  text-transform: lowercase;
  color: ${({ color = "currentcolor" }) => color};
`;

export const Logo: FunctionComponent<Logo> = props => (
  <Text {...props}>
    ùå∫ <Box display={["none", "inline"]}>analogic.al</Box>
  </Text>
);

export default Logo;
