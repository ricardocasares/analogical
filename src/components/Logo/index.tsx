import styled from "@emotion/styled";
import { FunctionComponent as F } from "react";
import { Box } from "@/components/Box";

export interface Logo {
  color?: string;
}

const Text = styled.span<Logo>`
  font-weight: 500;
  text-transform: lowercase;
  color: ${({ color = "currentcolor" }) => color};
`;

export const Logo: F<Logo> = (props) => (
  <Text {...props}>
    ùå∫ <Box display={["none", "inline"]}>analogic.al</Box>
  </Text>
);
