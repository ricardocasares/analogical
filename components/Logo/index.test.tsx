import React from "react";
import renderer from "react-test-renderer";
import serializer from "jest-emotion";
import Logo from "@app/components/Logo";

expect.addSnapshotSerializer(serializer);

test("renders with correct styles", () => {
  expect(renderer.create(<Logo />).toJSON()).toMatchSnapshot();
});
