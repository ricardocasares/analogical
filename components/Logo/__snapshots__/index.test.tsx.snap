// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with correct styles 1`] = `
.emotion-1 {
  font-weight: 500;
  text-transform: lowercase;
  color: currentcolor;
}

.emotion-0 {
  display: none;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    display: inline;
  }
}

<span
  className="emotion-1"
>
  ùå∫ 
  <div
    className="emotion-0"
    display={
      Array [
        "none",
        "inline",
      ]
    }
  >
    analogic.al
  </div>
</span>
`;
